src/tests/ftest/checksum/basic_checksum.py

```diff
@@ -0,0 +1,153 @@
+        self.pool.connect(2)
+        self.container.open()
+        self.obj.open()
+        self.ioreq = IORequest(self.context,
+                               self.container,
+                               self.obj,
+                               objtype=4)
+
+    def test_single_object_with_checksum(self):
+        """
+        Test ID: DAOS-3927
+        Test Description: Write Avocado Test to verify single data after
+                          pool/container disconnect/reconnect.
+        :avocado: tags=all,full_regression,pr,basic_checksum_object
+        """
+        self.d_log.info("Writing the Single Dataset")
+        record_index = 0
+        transaction = []
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0])
+                          * self.record_length[record_index])
+                c_dkey = ctypes.create_string_buffer("dkey {0}".format(dkey))
+                c_akey = ctypes.create_string_buffer("akey {0}".format(akey))
+                c_value = ctypes.create_string_buffer(indata)
+                c_size = ctypes.c_size_t(ctypes.sizeof(c_value))
+
+                new_transaction = self.container.get_new_tx()
+                self.ioreq.single_insert(c_dkey, c_akey, c_value, c_size,
+                                         new_transaction)
+                self.container.commit_tx(new_transaction)
+                transaction.append(new_transaction)
+                record_index = record_index + 1
+                if record_index == len(self.record_length):
+                    record_index = 0
+
+        self.reconnect()
+
+        self.d_log.info("Single Dataset Verification -- Started")
+        record_index = 0
+        transaction_index = 0
+        for dkey in range(self.no_of_dkeys):
+            for akey in range(self.no_of_akeys):
+                indata = ("{0}".format(str(akey)[0]) *
+                          self.record_length[record_index])
+                c_dkey = ctypes.create_string_buffer("dkey {0}".format(dkey))
+                c_akey = ctypes.create_string_buffer("akey {0}".format(akey))
+                val = self.ioreq.single_fetch(c_dkey,
+                                              c_akey,
+                                              len(indata)+1)
+                if indata != (repr(val.value)[1:-1]):
+                    self.d_log.error("ERROR:Data mismatch for "
+                                     "dkey = {0}, "
+                                     "akey = {1}".format(
+                                         "dkey {0}".format(dkey),
+                                         "akey {0}".format(akey)))
+                    self.fail("ERROR: Data mismatch for dkey = {0}, akey={1}"
+                              .format("dkey {0}".format(dkey),
+                                      "akey {0}".format(akey)))
+
+                transaction_index = transaction_index + 1
```

Comment:
How is that used?

Reply:
Removed the transaction_index... It is an unused variable. Looks like when I trimmed the original code, it was a left over.

Reply:
Done.